Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -113,15 +113,22 @@ define([
      *                  be treated as if their alpha is 1.0 everywhere.  When this property is false, memory usage
      *                  and texture upload time are potentially reduced.
      * @param {GetFeatureInfoFormat[]} [options.getFeatureInfoFormats] The formats in which to get feature information at a
-     *                                 specific location when {@see UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
+     *                                 specific location when {@link UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
      *                                 parameter is not specified, feature picking is disabled.
-     *
+     * @param {Boolean} [options.deferReadiness=false] True to defer full construction of this instance until a later call to {@link UrlTemplateImageryProvider#reinitialize}.
```

Comment:
No worries @tiffanylu, Promises can take a while to get accustomed to, so I'll try and lay out what I think is needed here.

1. Around line 100 in `createTileMapServiceImageryProvider` call `var deferred = when.defer()` to create a deferred promise, then pass `deferred.promise` as options object into `UrlTemplateImageryProvider`.
2. Replace calls to `imageryProvider.reinitialize` with `deferred.resolve`. passing in the same options.
3. In `UrlTermplateProvider` constructor, assign the `when(option).then` call to `this._readyPromise`.
4. In `UrlTermplateProvider`, you can set `ready` to true at the end of the constructor, while still inside of the `when(option).then` function.

The end result is that `UrlTermplateProvider` will expose the `readyPromise` that resolves right after `ready` is set to true and after all of the initial fetches have happened.

Hopefully that answers all of your questions, but if not let me know.  Thanks again for making the changes and sorry I didn't get a chance to review until today.